apps/www/_blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,189 @@
+---
+title: 'Realtime Broadcast and Presence Authorization'
+description: 'Create Authorization policies for Realtime Broadcast and Presence'
+author: filipe
+image: 2024-07-01-supabase-realtime-broadcast-and-presence-authorization/OG.png
+thumb: 2024-07-01-supabase-realtime-broadcast-and-presence-authorization/thumb.png
+categories:
+tags:
+  - realtime
+  - engineering
+date: '2024-07-01'
+toc_depth: 2
+---
+
+Today we're releasing Realtime Authorization for Broadcast and Presence in Public Alpha.
+
+For context, Supabase includes three useful extensions for building real-time applications.
+
+1. [Broadcast](https://supabase.com/docs/guides/realtime/broadcast): Send ephemeral, low-latency messages between users.
+2. [Presence](https://supabase.com/docs/guides/realtime/presence): Show when users are online and share state between users.
+3. [Postgres Changes](https://supabase.com/docs/guides/realtime/postgres-changes): Listen to Postgres database changes.
+
+This release introduces authorization for Broadcast and Presence using Row Level Security policies:
+
+```sql
+create policy "authenticated user can listen to all messages"
+on "realtime"."messages"
+as permissive for select
+to authenticated
+using ( true );
+```
+
+The `realtime.messages` table is created automatically by Supabase Realtime. It represents your [Channels](/docs/guides/realtime/concepts#channels) as objects against which you can write RLS policies. We test for a user's ability to `SELECT` and `INSERT` into this table to determine whether they can join, read, or post to a Realtime Channel.
+
+This approach of using RLS policies to control access should be familiar to users of [Supabase Storage](/docs/guides/storage/security/access-control).
+
+## How it works without Authorization
+
+When you want to connect to a Realtime Channel, you do something close to the following code:
+
+```tsx
+// With authenticated user token
+client.realtime.setAuth(token)
+
+client.channel('channel')
+.subscribe((status: string, err: any) => {
+  status === 'SUBSCRIBED' && console.log('Connected')
+})
```

Comment:
```suggestion
  .subscribe((status: string, err: any) => {
    if (status === 'SUBSCRIBED') {
      console.log('Connected')
    }
  })
```

Lots of people have no idea that you can call functions like that and might be confused

Reply:
good call!!!